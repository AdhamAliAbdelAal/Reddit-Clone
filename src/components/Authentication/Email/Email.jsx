import { useState } from 'react';

// Compoenets
import ReCAPTCHA from 'react-google-recaptcha';

// styles
import {
  FirstPartyContainer, RedditTextField, RedditLoadingButton,
} from '../styles';

// scripts
import { checkEmail } from '../scripts';

/**
 * Email Compoenet with ReCAPTCHA if required
 *
 * @component
 * @property {object} --email email Taken from input field
 * @property {function} --setEmail set email object
 * @property {function} --onSubmitFn On Submit form
 * @property {boolean} --loading loading prop ofr RedditLoadingButton
 * @property {string} --width width of the form
 * @property {string} --buttonText text on the Submit button
 * @property {string} --fieldText text on the input field
 * @property {string} --btnWidth submit button width
 * @property {bool} --recaptcha if we ReCAPTCHA is required on not
 * @property {function} --setVerified to set that recaptcha is verified
 * @property {boolean} --disabled to disabled prop ofr RedditLoadingButton
 * @returns {React.Component} Email Form
 */
function Email({
  email, setEmail, onSubmitFn, loading, width, buttonText, fieldText, btnWidth, recaptcha, setVerified, disabled,
}) {
  const [defaultEmailValue, setdefaultEmailValue] = useState(email?.input);
  const [recaptchaState, setrecaptchaState] = useState(false);

  return (
    // right value emailon submit in case that we have made any change in input field
    // in case no change the value there is wrong :) but the view here is true
    <FirstPartyContainer width={width} onSubmit={(e) => { e.preventDefault(); checkEmail(email.input, setEmail); onSubmitFn(); }} noValidate data-testid="SignUpEmail-test">
      <RedditTextField
        label={fieldText}
        variant="filled"
        required
        InputProps={{
          endAdornment: (
            email?.icon
          ),
          disableUnderline: true,
        }}
        clr={email?.color}
        onBlur={() => { if (recaptcha) setrecaptchaState(true); }}
        onChange={(e) => {
          setEmail((prevState) => ({
            ...prevState,
            input: e.target.value.trim(),
          }));
          checkEmail(e.target.value.trim(), setEmail);
          setdefaultEmailValue(e.target.value.trim());
        }}
        helperText={email?.error}
        value={defaultEmailValue || ''}
      />
      <RedditLoadingButton type="submit" loading={loading} data-testid="email-btn-test" width={btnWidth} disabled={disabled}>
        {buttonText}
      </RedditLoadingButton>
      {recaptchaState ? (
        <ReCAPTCHA
          sitekey="6LdjH-kiAAAAANFbV6SUnCjXNK3Z0h7q7j4IFf7i"
          onExpired={() => setVerified(false)}
          onChange={() => setVerified(true)}
          size="normal"
          sx={{ width: '320px' }}
        />
      )
        : null}
    </FirstPartyContainer>
  );
}
export default Email;
