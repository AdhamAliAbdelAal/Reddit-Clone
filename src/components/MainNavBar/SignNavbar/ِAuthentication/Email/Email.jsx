import { useState } from 'react';

// Compoenets
import ReCAPTCHA from 'react-google-recaptcha';

// styles
import {
  FirstPartyContainer, RedditTextField, RedditLoadingButton,
} from '../styles';

// scripts
import { checkEmail } from '../../../../Authentication/authenticationServer';

// env Variables
const { REACT_APP_SITEKEY } = process.env;

function Email({
  email, setEmail, onSubmitFn, loading, width, buttonText, fieldText, btnWidth, recaptcha, setVerified, disabled,
}) {
  const [defaultEmailValue, setdefaultEmailValue] = useState(email?.input);
  const [recaptchaState, setrecaptchaState] = useState(false);

  return (
    // right value emailon submit in case that we have made any change in input field
    // in case no change the value there is wrong :) but the view here is true
    <FirstPartyContainer width={width} onSubmit={(e) => { e.preventDefault(); checkEmail(email.input, setEmail); onSubmitFn(); }} noValidate>
      <RedditTextField
        label={fieldText}
        variant="filled"
        required
        InputProps={{
          endAdornment: (
            email?.icon
          ),
          disableUnderline: true,
        }}
        clr={email?.color}
        onBlur={() => { if (recaptcha) setrecaptchaState(true); }}
        onChange={(e) => {
          if (recaptcha) setrecaptchaState(true);
          setEmail((prevState) => ({
            ...prevState,
            input: e.target.value.trim(),
          }));
          checkEmail(e.target.value.trim(), setEmail);
          setdefaultEmailValue(e.target.value.trim());
        }}
        helperText={email?.error}
        value={defaultEmailValue || ''}
      />
      <RedditLoadingButton type="submit" loading={loading} width={btnWidth} disabled={disabled}>
        {buttonText}
      </RedditLoadingButton>
      {recaptchaState ? (
        <ReCAPTCHA
          sitekey={REACT_APP_SITEKEY}
          onExpired={() => setVerified(false)}
          onChange={() => setVerified(true)}
          size="normal"
          sx={{ width: '320px' }}
        />
      )
        : null}
    </FirstPartyContainer>

  );
}

export default Email;
