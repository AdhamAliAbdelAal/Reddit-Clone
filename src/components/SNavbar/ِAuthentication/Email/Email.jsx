import { useState } from 'react';

// Compoenets
import ReCAPTCHA from 'react-google-recaptcha';

// styles
import {
  FirstPartyContainer, RedditTextField, RedditLoadingButton,
} from '../styles';

// scripts
import { checkEmail } from '../../../Authentication/scripts';

function Email({
  email, setEmail, onSubmitFn, loading, width, buttonText, fieldText, btnWidth, recaptcha, setVerified, disabled, ispopup,
}) {
  const [defaultEmailValue, setdefaultEmailValue] = useState(email?.input);
  const [recaptchaState, setrecaptchaState] = useState(false);

  return (
    // right value emailon submit in case that we have made any change in input field
    // in case no change the value there is wrong :) but the view here is true
    <FirstPartyContainer width={width} onSubmit={(e) => { e.preventDefault(); checkEmail(email.input, setEmail); onSubmitFn(); }} noValidate>
      <RedditTextField
        label={fieldText}
        variant="filled"
        required
        InputProps={{
          endAdornment: (
            email?.icon
          ),
          disableUnderline: true,
        }}
        clr={email?.color}
        onBlur={() => { if (recaptcha) setrecaptchaState(true); }}
        onChange={(e) => {
          if (recaptcha) setrecaptchaState(true);
          setEmail((prevState) => ({
            ...prevState,
            input: e.target.value.trim(),
          }));
          checkEmail(e.target.value.trim(), setEmail);
          setdefaultEmailValue(e.target.value.trim());
        }}
        helperText={email?.error}
        value={defaultEmailValue || ''}
        ispopup={ispopup}
      />
      <RedditLoadingButton type="submit" loading={loading} width={btnWidth} disabled={disabled}>
        {buttonText}
      </RedditLoadingButton>
      {recaptchaState ? (
        <ReCAPTCHA
          sitekey="6LdjH-kiAAAAANFbV6SUnCjXNK3Z0h7q7j4IFf7i"
          onExpired={() => setVerified(false)}
          onChange={() => setVerified(true)}
          size="normal"
          sx={{ width: '320px' }}
        />
      )
        : null}
    </FirstPartyContainer>

  );
}

export default Email;
