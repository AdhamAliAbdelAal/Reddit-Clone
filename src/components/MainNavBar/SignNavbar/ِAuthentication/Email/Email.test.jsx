// import {
//   screen, render, fireEvent,
// } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Email from './Email';

// it('Invlaid Email', () => {
//   render(<Email />);
//   // Invalid Email Format
//   const field1 = screen.getByTestId('SignUpEmail-test').querySelector('input');
//   expect(field1).toBeInTheDocument();
//   fireEvent.change(field1, { target: { value: 'Bamsa_Elhoseny' } });

//   const emailbtn = screen.getByTestId('email-btn-test');
//   expect(emailbtn).toBeInTheDocument();
//   fireEvent.click(emailbtn);

//   // The Email isn't Valid
//   expect(screen.getByText('Please fix your email to continue')).toBeInTheDocument();
//   expect(screen.getByTestId('SignUpEmail-test')).toBeInTheDocument();
// });

// it('Valid Email', () => {
//   render(<Email />);
//   // Invalid Email Format
//   const field1 = screen.getByTestId('SignUpEmail-test').querySelector('input');
//   expect(field1).toBeInTheDocument();
//   fireEvent.change(field1, { target: { value: 'basmaelhoseny6@gmail.com' } });

//   const emailbtn = screen.getByTestId('email-btn-test');
//   expect(emailbtn).toBeInTheDocument();
//   fireEvent.click(emailbtn);

//   // The Email is Valid
//   expect(screen.getByTestId('SignUpEmail-test')).not.toBeInTheDocument();
// });

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<Email />).toJSON();
  expect(tree).toMatchSnapshot();
});
