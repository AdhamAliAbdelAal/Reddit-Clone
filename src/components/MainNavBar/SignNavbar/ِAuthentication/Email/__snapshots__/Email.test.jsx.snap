// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<form
  className="css-1b9eai0"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root css-5ngv9u-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-ut38f7-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    disabled={false}
    id=":r1:"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  />
</form>
`;
